JSONObject.java

```diff
@@ -578,10 +581,10 @@ public double getDouble(String key) throws JSONException {
         Object object = this.get(key);
         try {
             return object instanceof Number ? ((Number) object).doubleValue()
-                    : Double.parseDouble((String) object);
+                    : new BigDecimal((String) object).doubleValue();
```

Comment:
Which you then convert to `double`, good job then truncating it into a different number.